src/main/java/com/es/phoneshop/web/DeleteCartItemServlet.java

```diff
@@ -0,0 +1,38 @@
+package com.es.phoneshop.web;
+
+import com.es.phoneshop.model.cart.Cart;
+import com.es.phoneshop.model.cart.CartService;
+import com.es.phoneshop.model.cart.DefaultCartService;
+import com.es.phoneshop.model.product.ArrayListProductDao;
+import com.es.phoneshop.model.product.ProductDao;
+
+import javax.servlet.ServletConfig;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+public class DeleteCartItemServlet extends HttpServlet {
+    private ProductDao productDao;
+    private CartService cartService;
+
+    @Override
+    public void init(ServletConfig config) throws ServletException {
+        super.init(config);
+        productDao = ArrayListProductDao.getInstance();
+        cartService = DefaultCartService.getInstance();
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String productId = request.getPathInfo().substring(1);
+
+        Cart cart = cartService.getCart(request);
+        cartService.delete(cart, Long.valueOf(productId));
```

Comment:
We still need to add validation on back-end: for the valid number ID existing product.
Otherwise, the user can alter HTML code to pass the arbitrary product ID in a method call.